group-coordinator/src/main/java/org/apache/kafka/coordinator/group/generic/GenericGroupMember.java

```diff
@@ -0,0 +1,499 @@
+     */
+    public String clientId() {
+        return clientId;
+    }
+
+    /**
+     * @return the client host.
+     */
+    public String clientHost() {
+        return clientHost;
+    }
+
+    /**
+     * @return the rebalance timeout in milliseconds.
+     */
+    public int rebalanceTimeoutMs() {
+        return rebalanceTimeoutMs;
+    }
+
+    /**
+     * @return the session timeout in milliseconds.
+     */
+    public int sessionTimeoutMs() {
+        return sessionTimeoutMs;
+    }
+
+    /**
+     * @return the protocol type.
+     */
+    public String protocolType() {
+        return protocolType;
+    }
+
+    /**
+     * @return the list of supported protocols.
+     */
+    public List<Protocol> supportedProtocols() {
+        return supportedProtocols;
+    }
+
+    /**
+     * @return the member's assignment.
+     */
+    public byte[] assignment() {
+        return assignment;
+    }
+
+    /**
+     * @return the awaiting join callback.
+     */
+    public CompletableFuture<JoinGroupResponseData> awaitingJoinCallback() {
+        return awaitingJoinCallback;
+    }
+
+    /**
+     * @param value the updated join callback.
+     */
+    public void setAwaitingJoinCallback(CompletableFuture<JoinGroupResponseData> value) {
+        this.awaitingJoinCallback = value;
+    }
+
+    /**
+     * @return the awaiting sync callback.
+     */
+    public CompletableFuture<SyncGroupResponseData> awaitingSyncCallback() {
+        return awaitingSyncCallback;
+    }
+
+    /**
+     * @param value the updated sync callback.
+     */
+    public void setAwaitingSyncCallback(CompletableFuture<SyncGroupResponseData> value) {
+        this.awaitingSyncCallback = value;
+    }
+
+    /**
+     * @return true if the member is new, false otherwise.
+     */
+    public boolean isNew() {
+        return isNew;
+    }
+
+    /**
+     * @param value true if the member is new, false otherwise.
+     */
+    public void setIsNew(boolean value) {
+        this.isNew = value;
+    }
+
+    /**
+     * @return true if the existing heartbeat was satisfied, false otherwise.
+     */
+    public boolean heartBeatSatisfied() {
+        return heartbeatSatisfied;
+    }
+
+    /**
+     * @param value whether the heartbeat was satisfied.
+     */
+    public void setHeartBeatSatisfied(boolean value) {
+        this.heartbeatSatisfied = value;
+    }
+
+    @Override
+    public String toString() {
+        return "GenericGroupMember(" +
+            "memberId='" + memberId + '\'' +
+            ", groupInstanceId='" + groupInstanceId + '\'' +
+            ", clientId='" + clientId + '\'' +
+            ", clientHost='" + clientHost + '\'' +
+            ", rebalanceTimeoutMs=" + rebalanceTimeoutMs +
+            ", sessionTimeoutMs=" + sessionTimeoutMs +
+            ", protocolType='" + protocolType + '\'' +
+            ", supportedProtocols=" + supportedProtocols +
+            ')';
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Do we actually need `equals` and `hashCode`?

Reply:
i added it as i was following the builder pattern. let me remove it